File: 06a.pl
#!/usr/bin/perl
use warnings;
use strict;
use feature qw{ say };

use ARGV::OrDATA;

my (undef, @times)     = split ' ', <>;
my (undef, @distances) = split ' ', <>;

my $score = 1;
for my $race (0 .. $#times) {
    my $wins = grep $_ > $distances[$race],
               map +($times[$race] - $_) * $_,
               1 .. $times[$race] - 1;
    $score *= $wins;
}

say $score;

__DATA__
Time:      7  15   30
Distance:  9  40  200

File: 06a.py
#! /usr/bin/python3
import sys

def main(file_name: str):
    with open(file_name, 'r') as fh:
        times = list(map(int, (fh.readline().split())[1:]))
        distances = list(map(int, (fh.readline().split())[1:]))

        score = 1
        for i, time in enumerate(times):
            wins = 0
            for hold in range(time - 1):
                s = hold * (time - hold)
                if s > distances[i]:
                    wins += 1
            score *= wins
        print(score)

main(sys.argv[1])
